src/server/channel_store.cc

```diff
@@ -146,7 +146,8 @@ unsigned ChannelStore::SendMessages(std::string_view channel, facade::ArgRange m
     auto it = lower_bound(subscribers_ptr->begin(), subscribers_ptr->end(), idx,
                           ChannelStore::Subscriber::ByThreadId);
     while (it != subscribers_ptr->end() && it->Thread() == idx) {
-      if (auto* ptr = it->Get(); ptr)
+      // if ptr->cntx() is null, a connection might have closed or be in the process of closing
```

Comment:
Can it be `not null` yet the state `is_closing` ? If thast's the case shouldn't we also cover this here ?

Reply:
Sorry, I'm not sure which `is_closing` state you are referring to, there doesn't seem to be such a state in `facade::Connection` nor in `util::Connection`?

Reply:
when we close a connection we set the member field of `conn_context` called `conn_closing`. I am asking if we also need to poll that here?

Reply:
`ptr->cntx()->is_closing`

Reply:
Gotcha, `conn_closing` case is already supported and handled inside `SendAsync`, so we're good with that regards